cubical/data_handler/TiggerSourceProvider.py

```diff
@@ -96,48 +108,31 @@ def point_stokes(self, context):
         """ Returns a stokes parameter array to Montblanc. """
 
         # Get the extents of the time, baseline and chan dimension
-        (lt, ut), (lp, up) = context.dim_extents('ntime', 'npsrc')
-
+        (lp, up), (lt, ut), (lc, uc) = context.dim_extents('npsrc',
+                                                           'ntime',
+                                                           'nchan')
+        # (npsrc, ntime, nchan, 4)
         stokes = np.empty(context.shape, context.dtype)
 
-        for ind, source in enumerate(self._pnt_sources[lp:up]):
-            stokes[ind,:,0] = source.flux.I
-            stokes[ind,:,1] = source.flux.Q
-            stokes[ind,:,2] = source.flux.U
-            stokes[ind,:,3] = source.flux.V
-
-        return stokes
-
-    def point_alpha(self, context):
-        """ Returns a spectral index (alpha) array to Montblanc. """
-
-        alpha = np.empty(context.shape, context.dtype)
-
-        (lp, up) = context.dim_extents('npsrc')
+        f = self._freqs[None, lc:uc]
 
         for ind, source in enumerate(self._pnt_sources[lp:up]):
             try:
-                alpha[ind] = source.spectrum.spi
-            except:
-                alpha[ind] = 0
+                rf = source.spectrum.freq0
+            except AttributeError:
+                # TODO(jkenyon, osmirnov)
+                # I think this should be "or 1"
+                rf = self._sm.freq0 or 0
 
-        return alpha
-
-    def point_ref_freq(self, context):
-        """ Returns a reference frequency per source array to Montblanc. """
-        
-        pt_ref_freq = np.empty(context.shape, context.dtype)
-
-        (lp, up) = context.dim_extents('npsrc')
-        
-        for ind, source in enumerate(self._pnt_sources[lp:up]):
-            try:
-                pt_ref_freq[ind] = source.spectrum.freq0
-            except:
-                pt_ref_freq[ind] = self._sm.freq0 or 0
+            a = source.spectrum.spi
+            spectrum = (f/rf)**a
 
-        return pt_ref_freq
+            stokes[ind, :, :, 0] = source.flux.I[None, None]*spectrum
```

Comment:
Falling over here with:
```
  File "/home/oms/projects/CubiCal/cubical/data_handler/TiggerSourceProvider.py", line 134, in point_stokes
    stokes[ind, :, :, 0] = source.flux.I[None, None]*spectrum
ValueError: An exception occurred while obtaining data from data source 'point_stokes'

'float' object has no attribute '__getitem__'
```
I've removed the ``[None, None]`` indexing -- @sjperkins what was the idea of putting it in?

Reply:
To broadcast it up to the full (source, time, chan, corr) dimensionality.

Reply:
But ``source.flux.I`` is a float. It's a Tigger LSM source attribute -- the LSM doesn't have specific times or channels.

Reply:
Sure, its just the most general  (and admittedly ugly) way of handling the general case of both scintillation and spectra, even when they are not required.

Reply:
But it doesn't work -- it falls over with an exception, see above. Shouldn't we convert the floats into 0-D numpy arrays, in the interest of generality?

Does the ``ddfacet`` branch know about linear vs. circular feeds? I'm testing this now, and I'm pretty sure the predict is using XY feeds where I would expect RL.

Reply:
@sjperkins any comment on XY vs RL?